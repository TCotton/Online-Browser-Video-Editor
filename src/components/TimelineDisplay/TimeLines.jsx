import React from 'react'
import {useSelector} from "react-redux";
import ErrorBoundary from "../ErrorBoundary/Error";
import {imgArray} from '../slices/imageSlice';
import {data} from '../slices/waveformSlice';
import {ImageFramesComponent} from "./ImageFramesComponent";
import {D3WavelineComponent} from "./D3WavelineComponent";
import {TimeDisplay} from "./TimeDisplay";
import {D3LineTick} from './D3LineTick';
import {time} from "../slices/videoSlice";

export const TimeLines = () => {

    const iArr = useSelector(imgArray);
    const waveForm = useSelector(data);
    const timeDisplay = useSelector(time);

    return (
        <section className="bottom" data-testid="TimeLines">
            <div className="row-timeline">
                <div className="controls">
                </div>
                <div className="timeline">
                    <ErrorBoundary>
                        <D3LineTick data={waveForm} time={timeDisplay} height={30} width={807} />
                    </ErrorBoundary>
                </div>
            </div>
            <div className="row-one">
                <div className="controls">
                    <button>
                        <img className="eye" src={"./eye.svg"} alt=""/>
                    </button>
                </div>
                <div className="timeline">
                    <ErrorBoundary>
                        <ImageFramesComponent images={iArr}/>
                    </ErrorBoundary>
                </div>
            </div>
            <div className="row-two">
                <div className="controls">
                    <button>
                        <img className="eye" src={"./eye.svg"} alt=""/>
                    </button>
                </div>
                <div className="timeline">
                    <ErrorBoundary>
                        <D3WavelineComponent data={waveForm} height={98} width={807}/>
                    </ErrorBoundary>
                </div>
            </div>
            <div className="row-three">
                <div className="controls"/>
                <div className="timeline">
                    <ErrorBoundary>
                        <TimeDisplay/>
                    </ErrorBoundary>
                </div>
            </div>
        </section>
    )
}