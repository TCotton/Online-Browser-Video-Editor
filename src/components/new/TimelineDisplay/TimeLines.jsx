import React from 'react'
import {useSelector} from "react-redux";
import ErrorBoundary from "../../ErrorBoundary/Error";
import {imgArray} from '../slices/imageSlice';
import {data} from '../slices/waveformSlice';
import {ImageFramesComponent} from "./ImageFramesComponent";
import {WavelineComponent} from "./WavelineComponent";

export const TimeLines = () => {

    const iArr = useSelector(imgArray);
    const waveForm = useSelector(data);

    return (
        <section className="bottom">
            <div className="row-one">
                <div className="controls">
                    <button>
                        <img className="eye" src={"./eye.svg"} alt=""/>
                    </button>
                </div>
                <div className="timeline">
                    <ErrorBoundary>
                        <ImageFramesComponent images={iArr}/>
                    </ErrorBoundary>
                </div>
            </div>
            <div className="row-two">
                <div className="controls">
                    <button>
                        <img className="eye" src={"./eye.svg"} alt=""/>
                    </button>
                </div>
                <div className="timeline">
                    <ErrorBoundary>
                        <WavelineComponent data={waveForm} height={500} width={100} />
                    </ErrorBoundary>
                </div>
            </div>
        </section>
    )
}