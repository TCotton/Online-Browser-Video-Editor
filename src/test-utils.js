import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { createStore, combineReducers } from 'redux'
import { Provider } from 'react-redux'

import counterReducer from './components/slices/counterSlice';
import filesReducer from './components/slices/filesSlice';
import videoReducer from './components/slices/videoSlice';
import playerReducer from './components/slices/playerSlice';
import audioReducer from './components/slices/audioSlice';
import imageReducer from './components/slices/imageSlice';
import waveformReducer from './components/slices/waveformSlice';

function render(
    ui,
    {
        initialState,
        store = createStore(combineReducers({
            counter: counterReducer,
            files:  filesReducer,
            video: videoReducer,
            player: playerReducer,
            audio: audioReducer,
            images: imageReducer,
            waveform: waveformReducer,
        }), initialState),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {
        return <Provider store={store}>{children}</Provider>
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
// eslint-disable-next-line import/no-extraneous-dependencies
export * from '@testing-library/react'
// override render method
export { render }