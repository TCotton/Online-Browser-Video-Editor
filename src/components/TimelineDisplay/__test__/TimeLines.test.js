import React from 'react'
import { TimeLines } from '../TimeLines'
import { render } from '../../../test-utils'

describe('TimeLines', () => {
  it('should render successfully', () => {
    const { container } = render(<TimeLines />, {
      initialState: {
        video: {
          time: 999
        },
        waveform: {
          data: {
            min_array: [0],
            max_array: [0],
            length: 0
          }
        },
        images: {
          iArray: ['img']
        }
      }
    })
    expect(container.querySelector('[data-testid="TimeLines"]')).not.toBeNull()
  })

  it('should match snapshot', () => {
    const { baseElement } = render(<TimeLines />, {
      initialState: {
        video: {
          time: 999
        },
        waveform: {
          data: {
            min_array: [0],
            max_array: [0],
            length: 0
          }
        },
        images: {
          iArray: ['img']
        }
      }
    })
    expect(baseElement).toMatchSnapshot()
  })
})
