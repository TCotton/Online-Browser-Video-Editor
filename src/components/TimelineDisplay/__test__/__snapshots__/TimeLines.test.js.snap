// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeLines should match snapshot 1`] = `
<body>
  <div>
    <section
      class="bottom"
      data-testid="TimeLines"
    >
      <div
        class="row-timeline"
      >
        <div
          class="controls"
        />
        <div
          class="timeline"
        >
          <div
            class="timeline-ticks"
          >
            <svg
              data-testid="svg"
              preserveAspectRatio="xMidYMid slice"
              viewBox="0 0 807 30"
            >
              <g
                class="axis-top"
                data-testid="axis"
                fill="none"
                font-family="sans-serif"
                font-size="10"
                text-anchor="middle"
                transform="translate(0,0)"
              >
                <path
                  class="domain"
                  d="M0.5,-6V0.5H807.5V-6"
                  stroke="currentColor"
                />
                <g
                  class="tick"
                  opacity="1"
                  transform="translate(404,0)"
                >
                  <line
                    stroke="currentColor"
                    y2="-6"
                  />
                  <text
                    dy="0em"
                    fill="currentColor"
                    style="text-anchor: end; font-size: 1.75ex; font-family: monospace; fill: #ffffff;"
                    transform="translate(10,30) rotate(45)"
                    y="-9"
                  />
                </g>
              </g>
              <g
                class="waveform"
                data-testid="waveform"
              >
                <path
                  class="svg-line"
                  d="M403.5,15Z"
                  fill="#ffffff"
                />
              </g>
            </svg>
          </div>
        </div>
      </div>
      <div
        class="row-one"
      >
        <div
          class="controls"
        >
          <button>
            <img
              alt=""
              class="eye"
              src="./eye.svg"
            />
          </button>
        </div>
        <div
          class="timeline"
        >
          <div
            class="image-container"
            data-testid="ImageFramesComponent"
          >
            <img
              alt=""
              class="images"
              src="img"
            />
          </div>
        </div>
      </div>
      <div
        class="row-two"
      >
        <div
          class="controls"
        >
          <button>
            <img
              alt=""
              class="eye"
              src="./eye.svg"
            />
          </button>
        </div>
        <div
          class="timeline"
        >
          <svg
            data-testid="svg"
            height="98"
            preserveAspectRatio="xMidYMid slice"
            viewBox="0 0 807 98"
            width="807"
          >
            <g
              class="area"
              data-testid="area"
              transform="translate(0,0)"
            >
              <path
                class="wave"
                d="M403.5,49L403.5,49Z"
                transform="translate(0,0)"
              />
            </g>
          </svg>
        </div>
      </div>
      <div
        class="row-three"
      >
        <div
          class="controls"
        />
        <div
          class="timeline"
        >
          <div
            class="timeline-display"
            data-testid="TimeDisplay"
          >
            <test-file-stub
              style="transform: translate(-17px, -230px);"
            />
          </div>
        </div>
      </div>
    </section>
  </div>
</body>
`;
